### ISSUE TITLE
json_v2 parser - unix_ns timestamp not correct

### ISSUE THREAD

**Issue by hansij66** [influxdata/telegraf#10405]
### Relevent telegraf.conf

```toml
...
  data_format = "json_v2"
  [[inputs.mqtt_consumer.json_v2]]
    measurement_name = "test"
    timestamp_path = "@this.timestamp"
    timestamp_format = "unix_ns"
```

### Logs from Telegraf

N/A

### System info

docker telegraf:latest (1.21.2)

### Steps to reproduce

1. MQTT payload: {"timestamp":1641667732350000000, .............}
2. InfluxDB timestamp: 1641667732350000128

There is a small variation in the influxdb timestamp compared to MQTT timestamp. 

(note, I also have an input.mqtt_consumer defined with data_format = "influx", which also receives a unix_ns timestamp; this timestamp is exactly forwarded to influxdb

### Expected behavior

1. MQTT payload: {"timestamp":1641667732350000000, .............}
2. InfluxDB timestamp: 1641667732350000000

### Actual behavior

1. MQTT payload: {"timestamp":1641667732350000000, .............}
2. InfluxDB timestamp: 1641667732350000128

**Comment by powersj** [comment]
@sspaink thoughts on this one?

I think this may be related to https://github.com/influxdata/telegraf/issues/7555

If I use this input json:

```json
{
    "test": [
        { "value": 0 },
        { "value": 1 }
    ],
    "timestamp": 1631202459121654321
}
```

with this config:

```toml
[agent]
precision = "1ns"

[[inputs.file]]
  files = ["data.json"]
  data_format = "json_v2"
  [[inputs.file.json_v2]]
    measurement_name = "test"
    timestamp_path = "timestamp"
    timestamp_format = "unix_ns"
    [[inputs.file.json_v2.object]]
        path = "test"

[[outputs.file]]
```

I get:

```s
test,host=ryzen value=0 1631202459121654272
test,host=ryzen value=1 1631202459121654272
```

The timestamp is not correctly read in the first place and values are lost.

**Comment by sspaink** [comment]
@hansij66 thanks for opening an issue I think I have a fix in the pr: #10413. Also thanks @powersj for looking into it and providing example results and configs, this was an issue similar to #7555.
